src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -730,6 +730,11 @@ private BoundExpression BoxReceiver(BoundExpression rewrittenReceiver, NamedType
                 case BoundKind.Local:
                     return localsMayBeAssignedOrCaptured || ((BoundLocal)expression).LocalSymbol.RefKind != RefKind.None;
 
+                case BoundKind.TypeExpression:
```

Comment:
>case BoundKind.TypeExpression: [](start = 16, length = 30)

was this a pre-existing bug?

Reply:
No, it is needed due to changes added here, where we use this on a receiver which can be either a type or an expression.

---
In reply to: [301817913](https://github.com/dotnet/roslyn/pull/37052#discussion_r301817913) [](ancestors = 301817913)